integration-tests/smoke/ccip/ccip_reorg_test.go

```diff
@@ -0,0 +1,675 @@
+		Data:         []byte("hello world"),
+		TokenAmounts: nil,
+		FeeToken:     common.HexToAddress("0x0"),
+		ExtraArgs:    nil,
+	})
+	l.
+		Info().
+		Str("messageID", hexutil.Encode(reorgingMsgEvent.Message.Header.MessageId[:])).
+		Uint64("messageBlockNumber", reorgingMsgEvent.Raw.BlockNumber).
+		Str("messageBlockHashBeforeReorg", reorgingMsgEvent.Raw.BlockHash.String()).
+		Msg("sent CCIP message that will get re-orged before getting finalized")
+
+	const reorgDepth = 7
+	var minChainBlockNumberBeforeReorg = reorgingMsgEvent.Raw.BlockNumber + reorgDepth - 1
+	// let reorgDepth - 1 blocks pass by before re-orging the message.
+	// This will effectively rewind the chain to a block where the message didn't exist.
+	require.Eventually(t, func() bool {
+		bn, err := reorgSourceClient.BlockNumber()
+		require.NoError(t, err)
+		l.Info().
+			Int64("blockNumber", bn).
+			Uint64("targetBlockNumber", minChainBlockNumberBeforeReorg).
+			Msg("Waiting for chain to progress above target block number")
+		return bn >= int64(minChainBlockNumberBeforeReorg) //nolint:gosec // this will not overflow in the current test
+	}, 1*time.Minute, 500*time.Millisecond, "timeout exceeded: chain did not progress above the target block number")
+
+	// Run reorg below finality depth
+	l.Info().
+		Uint64("messageBlockNumber", reorgingMsgEvent.Raw.BlockNumber).
+		Int("reorgDepth", reorgDepth).
+		Uint64("sourceChainSelector", reorgSourceSelector).
+		Msg("starting blockchain reorg on Simulated Geth chain")
+	err = reorgSourceClient.GethSetHead(reorgDepth)
+	require.NoError(t, err, "error starting blockchain reorg on Simulated Geth chain")
+
+	bnAfterReorg, err := reorgSourceClient.BlockNumber()
+	require.NoError(t, err, "error getting block number after reorg")
+
+	l.Info().Int64("blockNumberAfterReorg", bnAfterReorg).Msg("block number after reorg")
+
+	reorgingMsgEvent = testhelpers.TestSendRequest(t, e.Env, state, reorgSourceSelector, destChainSelector, false, router.ClientEVM2AnyMessage{
+		Receiver:     common.LeftPadBytes(state.Chains[destChainSelector].Receiver.Address().Bytes(), 32),
+		Data:         []byte("hello world"),
+		TokenAmounts: nil,
+		FeeToken:     common.HexToAddress("0x0"),
+		ExtraArgs:    nil,
+	})
+	l.
+		Info().
+		Str("blockHashAfterReorg", reorgingMsgEvent.Raw.BlockHash.String()).
+		Str("messageID", hexutil.Encode(reorgingMsgEvent.Message.Header.MessageId[:])).
+		Uint64("messageBlockNumber", reorgingMsgEvent.Raw.BlockNumber).
+		Str("messageBlockHash", reorgingMsgEvent.Raw.BlockHash.String()).
+		Msgf("re-sent CCIP message after the re-org")
+
+	nodeAPIs := dockerEnv.GetCLClusterTestEnv().ClCluster.NodeAPIs()
+	nonBootstrapCount := len(nodeAPIs) - 1
+	l.Info().Msgf("waiting for %d non-bootstrap nodes to NOT report finality violation on the logpoller, since re-org is less than finality", nonBootstrapCount)
+	gomega.NewWithT(t).Consistently(func() bool {
```

Comment:
Is it okay to use gomega? I know there is no `consistently` check in `require`. But I got a comment before on a PR that we are getting rid of gomega.

Reply:
We still use it a lot all over the codebase, should be OK. If we want to prevent it I believe we can add a lint rule.